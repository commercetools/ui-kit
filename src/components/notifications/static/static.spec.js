import React from 'react';
import { render } from 'react-testing-library';
import Static from './static';

describe('Static', () => {
  it('should pass data-attributes all way to the DOM', () => {
    const { container } = render(
      <Static data-foo="bar" type="info">
        A message
      </Static>
    );

    expect(container.querySelector("[data-foo='bar']")).toBeInTheDocument();
  });

  it.each(['error', 'info', 'success', 'warning'])(
    'should render notification of type `%s`',
    type => {
      const { container } = render(<Static type={type}>A message</Static>);

      expect(container).toMatchSnapshot();
    }
  );
});
