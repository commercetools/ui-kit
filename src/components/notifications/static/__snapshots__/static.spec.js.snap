// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Static should render notification of type \`error\` 1`] = `
<div>
  <div
    class="notification error"
  >
    <div
      class="iconContainer"
    >
      <test-file-stub
        classname="theme-white"
        height="24"
        width="24"
      />
    </div>
    <div
      class="content"
    >
      A message
    </div>
  </div>
</div>
`;

exports[`Static should render notification of type \`info\` 1`] = `
<div>
  <div
    class="notification info"
  >
    <div
      class="iconContainer"
    >
      <test-file-stub
        classname="theme-white"
        height="24"
        width="24"
      />
    </div>
    <div
      class="content"
    >
      A message
    </div>
  </div>
</div>
`;

exports[`Static should render notification of type \`success\` 1`] = `
<div>
  <div
    class="notification success"
  >
    <div
      class="iconContainer"
    >
      <test-file-stub
        classname="theme-white"
        height="24"
        width="24"
      />
    </div>
    <div
      class="content"
    >
      A message
    </div>
  </div>
</div>
`;

exports[`Static should render notification of type \`warning\` 1`] = `
<div>
  <div
    class="notification warning"
  >
    <div
      class="iconContainer"
    >
      <test-file-stub
        classname="theme-white"
        height="24"
        width="24"
      />
    </div>
    <div
      class="content"
    >
      A message
    </div>
  </div>
</div>
`;
